src/components/TokenSelector.tsx

```diff
@@ -252,6 +252,32 @@ interface Props {
 const TokenSelector: React.FC<Props> = ({ isDisabled, tokens, selected, onChange, tabIndex = 0 }) => {
   const options = useMemo(() => tokens.map(token => ({ token, value: token.symbol, label: token.name })), [tokens])
 
+  // isFocused is used to force the menu to remain open and give focus to the search input
+  const [isFocused, setIsFocused] = useState(false)
+
+  const onSelectChange = useCallback(
+    (selected: { token: TokenDetails }, { action }: ActionMeta): void => {
+      // When an option is chosen, give control back to react-select
+      setIsFocused(false)
+
+      if (selected && action === 'select-option' && 'token' in selected) {
+        onChange(selected.token)
+      }
+    },
+    [onChange],
+  )
+
+  // When the search input is focused, force menu to remain open
+  const onMenuInputFocus = useCallback(() => setIsFocused(true), [])
+
+  const onInputChange = useCallback((inputValue): void => {
+    // `inputValue` is an empty string on `Escape`.
+    // Setting isFocused to false hides the menu as desired
+    if (!inputValue) {
+      setIsFocused(false)
```

Comment:
This closes the whole menu when input is deleted, sure we want that?

Reply:
hmmmmmm I see.
I would want that, but not sure about everyone else =P
Will try to not do that.

Reply:
Done! Used `onKeyDown` instead.